src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -649,6 +669,35 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
         }
       }
 
+      if (isInlineClass) {
+        primaryConstructor?.let {
+          // Inline classes only have 1 constructor parameter
+          check(it.parameters.size == 1) {
+            "Inline class must have 1 parameter in constructor"
+          }
+          // Inline class must have public primary constructor
+          check(PRIVATE !in it.modifiers && INTERNAL !in it.modifiers) {
+            "Inline class must have a public primary constructor"
+          }
+        }
+        // Inline classes can only have 1 backing property
+        check(propertySpecs.size == 1) {
+          "Inline class must have a single read-only (val) property."
+        }
+        // Inline classes property must be immutable.
+        check(!propertySpecs.first().mutable) {
```

Comment:
We should allow declaring multiple properties, and then there will be no guarantee that `first()` returns the primary constructor property, so we'll have to query the property by the corresponding constructor param name